src/alcotest/alcotest.ml

```diff
@@ -96,12 +99,12 @@ module Unix_platform (M : Alcotest_engine.Monad.S) = struct
 
   let with_redirect fd_file fn =
     let* () = M.return () in
-    Fmt.(flush stdout) ();
-    Fmt.(flush stderr) ();
+    Fmt.(flush (safe_stdout ())) ();
+    Fmt.(flush (safe_stderr ())) ();
     before_test ~output:fd_file ~stdout ~stderr;
```

Comment:
What about the calls to `before_test` and `after_test`? They use a single static variable (per fd) in the C stubs. Seems not so thread-safe.